sdstore/sdstore.go

```diff
@@ -370,7 +370,7 @@ func (s *sdStore) checkForRetry(res *http.Response, err error) (bool, error) {
 	}
 
 	if res.StatusCode/100 != 2 {
-		return true, nil
```

Comment:
does it change requires further changes to all places where store cli is used ? for example will this cause cache bookends to fail on `404` ?

Reply:
404 is handled separately here: https://github.com/screwdriver-cd/store-cli/blob/master/sdstore/sdstore.go#L365

This is just to return the actual status code to user for any non-200 and non-404 requests after retry

Reply:
And for 404, store-cli is returning 404 to user. It's bookend decided not to fail the cmd by `store-cli xx || true`